src/plugins/saved_objects_management/public/management_section/saved_objects_table_page.tsx

```diff
@@ -65,35 +71,39 @@ const SavedObjectsTablePage = ({
     ]);
   }, [setBreadcrumbs]);
 
+  const ContextWrapper = spacesApi?.ui.components.SpacesContext || EmptyFunctionComponent;
+
   return (
-    <SavedObjectsTable
-      initialQuery={initialQuery}
-      allowedTypes={allowedTypes}
-      serviceRegistry={serviceRegistry}
-      actionRegistry={actionRegistry}
-      columnRegistry={columnRegistry}
-      taggingApi={taggingApi}
-      savedObjectsClient={coreStart.savedObjects.client}
-      indexPatterns={dataStart.indexPatterns}
-      search={dataStart.search}
-      http={coreStart.http}
-      overlays={coreStart.overlays}
-      notifications={coreStart.notifications}
-      applications={coreStart.application}
-      perPageConfig={itemsPerPage}
-      goInspectObject={(savedObject) => {
-        const { editUrl } = savedObject.meta;
-        if (editUrl) {
-          return coreStart.application.navigateToUrl(
-            coreStart.http.basePath.prepend(`/app${editUrl}`)
-          );
-        }
-      }}
-      canGoInApp={(savedObject) => {
-        const { inAppUrl } = savedObject.meta;
-        return inAppUrl ? Boolean(get(capabilities, inAppUrl.uiCapabilitiesPath)) : false;
-      }}
-    />
+    <ContextWrapper>
+      <SavedObjectsTable
```

Comment:
This context approach feels definitely better than the column loading thing we had before.

The only remark is that it is introducing more tight/explicit coupling with the `spaces` plugin. This is fine for now, but if other 'providers' were to have the same need, we would ideally find a more generic API to allow adding top-level providers to the SOM listing page.

Reply:
Yeah and now the flyout also uses the data that is loaded by the context, so the Spaces list is only fetched once when the table is first loaded.

I thought about that and it seemed overkill for this PR. Agree we could use something more scalable in the future if the need arises.